modules/ting/src/Search/TingSearchFieldFilter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * @file
+ * The TingSearchFieldFilter class.
+ */
+
+namespace Ting\Search;
+
+/**
+ * Class TingSearchFieldFilter
+ *
+ * Represents a provider-independent comparison between a field instance and a
+ * value using an operator.
+ *
+ * @package Ting\Search
+ */
+class TingSearchFieldFilter implements BooleanStatementInterface {
+
+  /**
+   * The field.
+   *
+   * @var string
+   */
+  protected $name;
+
+  /**
+   * The operator used to compare the field instance and value.
+   *
+   * @var string
+   */
+  protected $operator;
+
+  /**
+   * Field value.
+   *
+   * @var mixed
+   */
+  protected $value;
+
+
+  /**
+   * The logic operator used to evaluate the field evaluation against the
+   * previous statement.
+   *
+   * @see BooleanStatementInterface::OP_*
+   *
+   * @var string
+   */
+  protected $logicOperator;
+
+  /**
+   * TingSearchFieldFilter constructor.
+   *
+   * @param string $name
+   *   The field name.
+   *
+   * @param mixed $value
+   *   Field value.
+   *
+   * @param string $operator
+   *   Operator to use when comparing the field instance with a value.
+   *
+   * @param string $logic_operator
+   *   Operator to use when comparing the evaluated field with a previous
+   *   statement.
+   */
+  public function __construct($name, $value, $operator = '=', $logic_operator = BooleanStatementInterface::OP_AND) {
```

Comment:
Should operators like `=` be enumerated as well?

Reply:
Might be a good idea, can't really decide whether it will be going a bit far or make the abstraction layer more robust - I'll punt it for now but will leave a comment.

Reply:
Please no. Using strings mimics the Drupal query classes (SQL/EntityFieldQuery), which is good.

Reply:
The current enum implementation is in essence stringly typed anyway. If we continue on that route I see no harm in at least having an operator enum and validating the input. This lets developers choose whether they want to use `Operator::EQUALS` or `=`.